src/main/java/net/pms/network/DbIdResourceLocator.java

```diff
@@ -121,39 +129,94 @@ public DLNAResource getDLNAResourceByDBID(DbidTypeAndIdent typeAndIdent) {
 						}
 						try (ResultSet resultSet = statement.executeQuery(sql)) {
-							res = new VirtualFolderDbId(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident, "");
+							res = new VirtualFolderDbId(typeAndIdent.ident,
+								new DbidTypeAndIdent(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident), "");
 							while (resultSet.next()) {
-								DLNAResource item = new RealFileDbId(DbidMediaType.TYPE_AUDIO, new File(resultSet.getString("FILENAME")),
-									resultSet.getString("FID"));
+								DLNAResource item = new RealFileDbId(
+									new DbidTypeAndIdent(DbidMediaType.TYPE_AUDIO, resultSet.getString("FID")),
+									new File(resultSet.getString("FILENAME")));
 								item.resolve();
 								res.addChild(item);
 							}
 						}
 						break;
-					case TYPE_PERSON:
+
+					case TYPE_PERSON_ALL_FILES:
 						sql = String.format(
-							"select FILENAME, F.ID as FID, MODIFIED from FILES as F left outer join AUDIOTRACKS as A on F.ID = A.FILEID where (  F.TYPE = 1  and  (A.ALBUMARTIST = '%s' or A.ARTIST = '%s'))",
+							"select FILENAME, F.ID as FID, MODIFIED from FILES as F left outer join AUDIOTRACKS as A on F.ID = A.FILEID " +
+								"where (A.ALBUMARTIST = '%s' or A.ARTIST = '%s')",
 							typeAndIdent.ident, typeAndIdent.ident);
 						if (LOGGER.isTraceEnabled()) {
 							LOGGER.trace(String.format("SQL PERSON : %s", sql));
 						}
 						try (ResultSet resultSet = statement.executeQuery(sql)) {
-							res = new VirtualFolderDbId(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident, "");
+							res = new VirtualFolderDbId(typeAndIdent.ident,
+								new DbidTypeAndIdent(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident), "");
+							while (resultSet.next()) {
+								DLNAResource item = new RealFileDbId(
+									new DbidTypeAndIdent(DbidMediaType.TYPE_AUDIO, resultSet.getString("FID")),
+									new File(resultSet.getString("FILENAME")));
+								item.resolve();
+								res.addChild(item);
+							}
+						}
+						res.setFakeParentId(encodeDbid(new DbidTypeAndIdent(DbidMediaType.TYPE_PERSON, typeAndIdent.ident)));
+						break;
+
+					case TYPE_PERSON:
+						res = new VirtualFolderDbId(typeAndIdent.ident, new DbidTypeAndIdent(DbidMediaType.TYPE_PERSON, typeAndIdent.ident),
+							"");
+						DLNAResource allFiles = new VirtualFolderDbId(Messages.getString("Search.AllFiles"),
+							new DbidTypeAndIdent(DbidMediaType.TYPE_PERSON_ALL_FILES, typeAndIdent.ident), "");
+						res.addChild(allFiles);
+						DLNAResource albums = new VirtualFolderDbId(Messages.getString("Search.ByAlbum"),
+							new DbidTypeAndIdent(DbidMediaType.TYPE_PERSON_ALBUM, typeAndIdent.ident), "");
+						res.addChild(albums);
+						break;
+
+					case TYPE_PERSON_ALBUM:
+						sql = String.format("SELECT DISTINCT(album) FROM AUDIOTRACKS A where COALESCE(A.ALBUMARTIST, A.ARTIST) = '%s'",
+							typeAndIdent.ident);
+						res = new VirtualFolderDbId(typeAndIdent.ident, new DbidTypeAndIdent(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident),
+							"");
```

Comment:
It's not something I would fail the review for but just FYI the formatting of the project, and what I think is more readable, would be this:

```suggestion
						res = new VirtualFolderDbId(
							typeAndIdent.ident,
							new DbidTypeAndIdent(DbidMediaType.TYPE_ALBUM, typeAndIdent.ident),
							""
						);
```